src/DevBetterWeb.Infrastructure/Services/DailyCheckPingService.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using DevBetterWeb.Core;
+using DevBetterWeb.Core.Entities;
+using DevBetterWeb.Core.Interfaces;
+using DevBetterWeb.Core.Specs;
+using System.Linq;
+using DevBetterWeb.Infrastructure.Identity.Data;
+using Microsoft.AspNetCore.Identity;
+using System;
+
+namespace DevBetterWeb.Infrastructure.Services
+{
+  public class DailyCheckPingService : IDailyCheckPingService
+  {
+    public IRepository<Invitation> _repository;
+    public IEmailService _emailService;
+    public UserManager<ApplicationUser> _userManager;
+
+    public DailyCheckPingService(IRepository<Invitation> repository,
+      IEmailService emailService,
+      UserManager<ApplicationUser> userManager)
+    {
+      _repository = repository;
+      _emailService = emailService;
+      _userManager = userManager;
+    }
+
+    public async Task SendPingIfNeeded(AppendOnlyStringList messages)
+    {
+      var spec = new ActiveInvitationsSpec();
+      var activeInvitations = await _repository.ListAsync(spec);
+
+      var invitationsForUserPing = CheckIfAnyActiveInvitationsRequireUserPing(activeInvitations);
+      if (invitationsForUserPing.Any())
+      {
+        foreach (var invitation in invitationsForUserPing)
+        {
+          messages.Append($"User at email {invitation.Email} should be reminded to finish setting up their account.");
+        }
+
+        var userPingMessages = await SendUserPing(invitationsForUserPing);
+        foreach (var message in userPingMessages)
+        {
+          messages.Append(message);
+        }
+      }
+
+      var invitationsForAdminPing = CheckIfAnyActiveInvitationsRequireAdminPing(activeInvitations);
+      if(invitationsForAdminPing.Any())
+      {
+        var listOfEmails = "";
+        foreach(var invitation in invitationsForAdminPing)
+        {
+          listOfEmails += $"{invitation.Email}\n";
+        }
+        messages.Append($"Admins should be reminded to remind these users to finish setting up their DevBetter accounts: {listOfEmails}");
+
+        var adminPingMessage = await SendAdminPing(invitationsForAdminPing);
+        messages.Append(adminPingMessage);
+      }
+    }
+
+    public List<Invitation> CheckIfAnyActiveInvitationsRequireAdminPing(List<Invitation> invitations)
+    {
+      var invitationsRequiringAdminPing = new List<Invitation>();
+
+      foreach(var invite in invitations)
+      {
+        bool inviteRequiresAdminPing = true;
+
+        if (invite.DateOfUserPing.Equals(DateTime.MinValue)) inviteRequiresAdminPing = false;
+        if (invite.DateOfLastAdminPing.Equals(DateTime.Today)) inviteRequiresAdminPing = false;
+        if (DateTime.Today - invite.DateCreated < new TimeSpan(4, 0, 0, 0)) inviteRequiresAdminPing = false;
+        if (DateTime.Today - invite.DateOfUserPing < new TimeSpan(2, 0, 0, 0)) inviteRequiresAdminPing = false;
```

Comment:
If you have magic numbers (4, 2) it's best to name them somewhere. If they're only used here, they should be named here. So you might say something like:

TimeSpan remind4DaysAfterCreation = new TimeSpan(4,0,0,0);
TimeSpan remind2DaysAfterLastPing = new TimeSpan(2,0,0,0);

Then your if statements would use these variables, not the magic timespan values directly.